front/src/app/modules/layout/register-page/register-page.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component } from '@angular/core';
+import {MatButtonModule} from "@angular/material/button";
+import {MatFormFieldModule} from "@angular/material/form-field";
+import {MatInputModule} from "@angular/material/input";
+import {NgIf} from "@angular/common";
+import {
+  AbstractControl, AsyncValidatorFn,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  ValidationErrors,
+  ValidatorFn,
+  Validators
+} from "@angular/forms";
+import {AuthService} from "../../../services/auth.service";
+import {Router} from "@angular/router";
+import {LoginCredentials, RegisterUserDto} from "../../../models/models";
+import {HttpErrorResponse} from "@angular/common/http";
+import {Observable, of} from "rxjs";
+import {UserService} from "../../../services/user.service";
+
+@Component({
+  selector: 'app-register-page',
+  standalone: true,
+    imports: [
+        MatButtonModule,
+        MatFormFieldModule,
+        MatInputModule,
+        NgIf,
+        ReactiveFormsModule
+    ],
+  templateUrl: './register-page.component.html',
+  styleUrl: './register-page.component.css'
+})
+export class RegisterPageComponent {
+  errorMessage: string = "";
+  registerForm:FormGroup = new FormGroup({
+    email: new FormControl('', [Validators.required, Validators.email]),
+    username: new FormControl('', [Validators.required]),
```

Comment:
max length validation?

Reply:
Fixed with most recent commits.

Reply:
Great! As far as I am concerned, you can merge.